algorithms/src/snark/marlin/ahp/prover/round_functions/second.rs

```diff
@@ -193,7 +194,16 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
                     // we rewrite this as  r_i * (z_a * (eta_c * z_b + 1) + eta_b * z_b);
                     // This is better since it reduces the number of required
                     // multiplications by `constraint_domain.size()`.
+                    //
+                    // LOOKUPS EDIT
+                    // turning this back into a simpler lincheck sumcheck to allow easy
+                    // modification of the rowcheck.
                     let mut summed_z_m = {
+                        cfg_iter_mut!(z_b.coeffs).for_each(|b| *b *= eta_b);
```

Comment:
I suspect we can change this into a single expression like

`let summed_z_m = cfg_iter!(z_a.coeffs).zip(z_b.coeffs).zip(z_c.coeffs).map(|((a, b), c)| a + eta_b * b + eta_c * c).collect();`

Reply:
So it turns out that `z_a` is actually 1 below the degree of `z_b` and`z_c`. Would simply pushing a `F::zero()` to the end of `z_a` work out in this case or could the degree mismatch fluctuate?

Reply:
I've added a bit of code that will just fill up `z_a` with zero coefficients until it matches `z_b`, though maybe this might not work in the non-ZK case (since the degree of `z_b` is otherwise just the size of the constraint domain)